File: yavel/src/androidTest/java/com/awolity/yavel/CryptUtilsTest.java
package com.awolity.yavel;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import java.util.HashSet;
import java.util.Set;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

public class CryptUtilsTest {

    private YavelKey yavelKey;

    @Before
    public void initKey()  throws Exception{
        yavelKey = new YavelKey("whatever");
    }

    @After
    public void deleteKey()  throws Exception{
        yavelKey.deleteKey();
    }

    @Test
    public void encryptString_decryptString() throws Exception {
        String plainString = "whatever1234";
        String encryptedString = CryptUtils.INSTANCE.encryptString(plainString, yavelKey.getKey());
        String decryptedString = CryptUtils.INSTANCE.decryptString(encryptedString, yavelKey.getKey());

        assertEquals(plainString, decryptedString);
    }

    @Test
    public void encryptString_decryptString2() throws Exception {
        String plainString = "whatever1234";
        String encryptedString = CryptUtils.INSTANCE.encryptString(plainString, yavelKey.getKey());

        yavelKey = new YavelKey("whatever2");

        try {
            @SuppressWarnings("UnusedAssignment") String decryptedString = CryptUtils.INSTANCE.decryptString(encryptedString, yavelKey.getKey());
        } catch (YavelException e) {
            assertEquals("javax.crypto.AEADBadTagException", e.getMessage());
        }
    }

    @Test
    public void encryptBoolean_decryptBoolean() throws Exception {
        String encryptedBoolean = CryptUtils.INSTANCE.encryptBoolean(true, yavelKey.getKey());
        boolean decryptedBoolean = CryptUtils.INSTANCE.decryptBoolean(encryptedBoolean, yavelKey.getKey());

        assertTrue(decryptedBoolean);
    }

    @Test
    public void encryptBoolean_decryptBoolean2() throws Exception {
        String encryptedBoolean = CryptUtils.INSTANCE.encryptBoolean(false, yavelKey.getKey());
        boolean decryptedBoolean = CryptUtils.INSTANCE.decryptBoolean(encryptedBoolean, yavelKey.getKey());

        assertFalse(decryptedBoolean);
    }

    @Test
    public void encryptFloat_decryptFloat() throws Exception {
        float value = (float) 6.836542E-8;
        String encryptedFloat = CryptUtils.INSTANCE.encryptFloat(value, yavelKey.getKey());
        float decryptedFloat = CryptUtils.INSTANCE.decryptFloat(encryptedFloat, yavelKey.getKey());

        assertEquals(value, decryptedFloat,0.0001F);
    }

    @Test
    public void encryptDouble_decryptDouble() throws Exception {
        double value = 6.836542E-8;
        String encryptedValue = CryptUtils.INSTANCE.encryptDouble(value, yavelKey.getKey());
        double decryptedValue = CryptUtils.INSTANCE.decryptFloat(encryptedValue, yavelKey.getKey());

        assertEquals(value, decryptedValue,0.0001F);
    }

    @Test
    public void encryptLong_decryptLong() throws Exception {
        long value = 654987321;
        String encryptedLong = CryptUtils.INSTANCE.encryptLong(value, yavelKey.getKey());
        float decryptedLong = CryptUtils.INSTANCE.decryptLong(encryptedLong, yavelKey.getKey());

        assertEquals(value, decryptedLong,1);
    }

    @Test
    public void encryptInt_decryptInt() throws Exception {
        int value = 654987;
        String encryptedInt = CryptUtils.INSTANCE.encryptInt(value, yavelKey.getKey());
        float decryptedInt = CryptUtils.INSTANCE.decryptInt(encryptedInt, yavelKey.getKey());

        assertEquals(value, decryptedInt,1);
    }
}

File: yavel/src/androidTest/java/com/awolity/yavel/CryptUtilsTest.kt
package com.awolity.yavel

import org.junit.After
import org.junit.Before
import org.junit.Test

import org.junit.Assert.assertEquals
import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue

class CryptUtilsTest {

    private var yavelKey: YavelKey? = null

    @Before
    @Throws(Exception::class)
    fun initKey() {
        yavelKey = YavelKey("whatever")
    }

    @After
    @Throws(Exception::class)
    fun deleteKey() {
        yavelKey!!.deleteKey()
    }

    @Test
    @Throws(Exception::class)
    fun encryptString_decryptString() {
        val plainString = "whatever1234"
        val encryptedString = CryptUtils.encryptString(plainString, yavelKey!!.key)
        val decryptedString = CryptUtils.decryptString(encryptedString, yavelKey!!.key)

        assertEquals(plainString, decryptedString)
    }

    @Test
    @Throws(Exception::class)
    fun encryptString_decryptString2() {
        val plainString = "whatever1234"
        val encryptedString = CryptUtils.encryptString(plainString, yavelKey!!.key)

        yavelKey = YavelKey("whatever2")

        try {
            val decryptedString = CryptUtils.decryptString(encryptedString, yavelKey!!.key)
        } catch (e: YavelException) {
            assertEquals("javax.crypto.AEADBadTagException", e.message)
        }
    }

    @Test
    @Throws(Exception::class)
    fun encryptBoolean_decryptBoolean() {
        val encryptedBoolean = CryptUtils.encryptBoolean(true, yavelKey!!.key)
        val decryptedBoolean = CryptUtils.decryptBoolean(encryptedBoolean, yavelKey!!.key)

        assertTrue(decryptedBoolean)
    }

    @Test
    @Throws(Exception::class)
    fun encryptBoolean_decryptBoolean2() {
        val encryptedBoolean = CryptUtils.encryptBoolean(false, yavelKey!!.key)
        val decryptedBoolean = CryptUtils.decryptBoolean(encryptedBoolean, yavelKey!!.key)

        assertFalse(decryptedBoolean)
    }

    @Test
    @Throws(Exception::class)
    fun encryptFloat_decryptFloat() {
        val value = 6.836542E-8.toFloat()
        val encryptedFloat = CryptUtils.encryptFloat(value, yavelKey!!.key)
        val decryptedFloat = CryptUtils.decryptFloat(encryptedFloat, yavelKey!!.key)

        assertEquals(value, decryptedFloat, 0.0001f)
    }

    @Test
    @Throws(Exception::class)
    fun encryptDouble_decryptDouble() {
        val value = 6.836542E-8
        val encryptedValue = CryptUtils.encryptDouble(value, yavelKey!!.key)
        val decryptedValue = CryptUtils.decryptFloat(encryptedValue, yavelKey!!.key).toDouble()

        assertEquals(value, decryptedValue, 0.0001)
    }

    @Test
    @Throws(Exception::class)
    fun encryptLong_decryptLong() {
        val value: Long = 654987321
        val encryptedLong = CryptUtils.encryptLong(value, yavelKey!!.key)
        val decryptedLong = CryptUtils.decryptLong(encryptedLong, yavelKey!!.key).toFloat()

        assertEquals(value.toFloat(), decryptedLong, 1f)
    }

    @Test
    @Throws(Exception::class)
    fun encryptInt_decryptInt() {
        val value = 654987
        val encryptedInt = CryptUtils.encryptInt(value, yavelKey!!.key)
        val decryptedInt = CryptUtils.decryptInt(encryptedInt, yavelKey!!.key).toFloat()

        assertEquals(value.toFloat(), decryptedInt, 1f)
    }
}
